dashboard/src/components/AppView/AppView.tsx

```diff
@@ -100,19 +124,27 @@ class AppView extends React.Component<IAppViewProps, IAppViewState> {
       }, {});
     this.setState({ otherResources });
 
-    const deployments = manifest.filter(d => d.kind === "Deployment");
-    const services = manifest.filter(d => d.kind === "Service");
-    const ingresses = manifest.filter(d => d.kind === "Ingress");
     const sockets: WebSocket[] = [];
-    for (const d of deployments) {
-      sockets.push(this.getSocket("deployments", d.apiVersion, d.metadata.name, newApp.namespace));
-    }
-    for (const svc of services) {
-      sockets.push(this.getSocket("services", svc.apiVersion, svc.metadata.name, newApp.namespace));
-    }
-    for (const i of ingresses) {
-      sockets.push(this.getSocket("ingresses", i.apiVersion, i.metadata.name, newApp.namespace));
-    }
+    manifest.forEach(i => {
+      switch (i.kind) {
+        case "Deployment":
+          sockets.push(
+            this.getSocket("deployments", i.apiVersion, i.metadata.name, newApp.namespace),
+          );
+          break;
+        case "Service":
+          sockets.push(this.getSocket("services", i.apiVersion, i.metadata.name, newApp.namespace));
+          break;
+        case "Ingress":
+          sockets.push(
+            this.getSocket("ingresses", i.apiVersion, i.metadata.name, newApp.namespace),
+          );
+          break;
+        case "Secret":
+          this.props.getResource("v1", "secrets", newApp.namespace, i.metadata.name);
```

Comment:
Shouldn't `getApp` be in charge of getting the Secrets related to that app too? I think that's what we do when we get service bindings, for example?